src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -778,6 +778,11 @@ private BoundExpression BindTupleExpression(TupleExpressionSyntax node, Diagnost
             if (hasNaturalType)
             {
                 tupleTypeOpt = TupleTypeSymbol.Create(node.Location, elements, locations, elementNamesArray, this.Compilation, node, diagnostics);
+
+                if (ShouldCheckConstraints)
```

Comment:
>if (ShouldCheckConstraints) [](start = 16, length = 27)

I don't think constraints check should be conditional in this scenario. Where would it be performed otherwise?